src/main/java/in/projecteka/consentmanager/user/TransactionPinService.java

```diff
@@ -58,16 +62,20 @@ private boolean isPinValid(String pin) {
                     if (!encoder.matches(pin, transactionPin.get().getPin())) {
                         return Mono.error(ClientError.transactionPinDidNotMatch());
                     }
-
                     return Mono.empty();
-                }).thenReturn(newToken(patientId));
+                }).thenReturn(newToken(patientId, scope));
     }
 
     @SneakyThrows
-    private Token newToken(String userName) {
+    private Token newToken(String userName, String scope) {
         int minutes = userServiceProperties.getTransactionPinTokenValidity() * 60 * 1000;
+        HashMap<String, Object> claims = new HashMap<>(1);
+        claims.put("sid" , UUID.randomUUID().toString());
```

Comment:
session id introduced to avoid storing the whole token.

Reply:
Yes